vnext/proposals/active/keyboard-reconcile-desktop.md

```diff
@@ -0,0 +1,148 @@
+
+| **TYPE** | **REQUIRED** | **PLATFORM** | **DEFAULT** |
+| --- | --- | --- | --- |
+| boolean | No | Windows, macOS | false |
+
+React Native core introduced `focusable` that does the same thing as acceptsKeyboardFocus in v0.62. In order to align and make this complete, the work here includes the following:
+- **COMPLETED** : Windows has added `focusable` and marked `acceptsKeyboardFocus` as deprecated in 0.62. 
+- macOS needs to do add `focusable` and mark `acceptsKeyboardFocus` as deprecated. Tracked by - [Issue#498](https://github.com/microsoft/react-native-macos/issues/498)
+- **COMPLETED** : Add focusable to Pressable ([windows Issue#5512](https://github.com/microsoft/react-native-windows/issues/5512), [macOS Issue#500](https://github.com/microsoft/react-native-macos/issues/500)). 
+- `focusable` is already supported in Android and exists in core. There is no hard upstreaming requirement, however – it would be good to implement this for iOS/iPadOS for completion.
+
+## focus, blur, onFocus, onBlur on View
+
+React Native core supports `.focus()`, `.blur()` methods and `onFocus()`, `onBlur()` callbacks on TextInput component. Windows and macOS implementations add the same methods and callbacks on View in addition to supporting them on TextInput. The work remaining for these include:
+- **COMPLETED** : The event bubbling behavior with .focus() and .blur() on View in macOS has some differences from Windows and w3c standards. The behavior needs to be evaluated and reconciled between the 2 platforms.
+- The microsoft/react-native fork has implementations for all the above methods and callbacks for iOS and Android as well. Once the event bubbling behavior above has been reconciled, these can be upstreamed to core. Tracked by [Issue#243](https://github.com/microsoft/react-native-macos/issues/243)
+- TextInput also includes an `isFocused()` method that returns true/false based on whether this TextInput currently has focus. It will be useful to add this to View to and upstream as a separate PR. Tee this up with the previous one. Tracked by - [windows Issue#5511](https://github.com/microsoft/react-native-windows/issues/5511), [macOS Issue#499](https://github.com/microsoft/react-native-macos/issues/499)
+- Add `.focus()`, `.blur()`, `onFocus()`, `onBlur()` and `isFocused()` to Pressable. Tracked by [windows Issue#5589](https://github.com/microsoft/react-native-windows/issues/5589), [macOS Issue#518](https://github.com/microsoft/react-native-macos/issues/518)
+
+## Default keyboarding behaviors on JS components
+
+Both Windows and macOS implementations have added appropriate keyboarding defaults to JS components like Button, Touchables. The implementation includes firing onPress() callbacks on Button and Touchable components when they are in focus and Enter/Space key is pressed. The work remaining for these include:
```

Comment:
One thing to note is that different element types on web respond differently to the Spacebar. For example, on the web, Spacebar will not trigger a `press` for elements that are links. Therefore, when a `Pressable` is used to render a link on web it shouldn't respond to Spacebar. Is there anything similar to this in WinUI?

(This brings up a broader question of how "onPress" is implemented. React Native currently infers `onPress` from the lifecycle of a pointer and its positioning, which introduces somewhat error-prone rectangle measurements during the gesture. On the web, we can associate `onPress` with the native "click" event, which is a separate high-level event from the pointer events that may also be triggered by other interactions, e.g., keyboard, screenreader. At some point we should discuss whether `click` events should be implemented in React Native using similar heuristics to the W3C event. Currently, the RN View `onClick` prop is non-standard and for keyboards only)

Reply:
It does appear that on Windows, Space will trigger "Click" on Hyperlinks. There is some well defined documentation on keyboard interactions we can refer to here- [Keyboard interactions](https://docs.microsoft.com/en-us/windows/uwp/design/input/keyboard-interactions#space-key)

Reply:
After second round of review, we are leaving this as spec'ed for Windows and macOS to match the OS native behaviors.It may be ok for react-native-web to not handle Space to invoke if that matches web behaviors.